pkg/cmd/whoami/whoami_test.go

```diff
@@ -0,0 +1,139 @@
+			args: args{
+				cfg: &config.Config{
+					AccessToken:  "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpYXQiOjE2NzAzMTA4MDQsImF1dGhfdGltZSI6MTY3MDMwNDQxOCwianRpIjoiOGEyNmUyN2MtYmM3Zi00YjVhLWFiN2QtZjdiOTJhMGFmZGE4IiwiaXNzIjoiaHR0cHM6Ly9zc28ucmVkaGF0LmNvbS9hdXRoL3JlYWxtcy9yZWRoYXQtZXh0ZXJuYWwiLCJhdWQiOiJyaG9hcy1jbGktcHJvZCIsInN1YiI6ImY6NTI4ZDc2ZmYtZjcwOC00M2VkLThjZDUtZmUxNmY0ZmUwY2U2Omh3b3JsZCIsInR5cCI6IkJlYXJlciIsImF6cCI6InJob2FzLWNsaS1wcm9kIiwic2Vzc2lvbl9zdGF0ZSI6IjlhOTA2YjkzLTM2NTAtNGQ4MS1hYTNkLTNkNDA5ZmYxYTllNiIsImFsbG93ZWQtb3JpZ2lucyI6WyJodHRwOi8vMTI3LjAuMC4xIiwiaHR0cDovL2xvY2FsaG9zdCIsImh0dHBzOi8vY29uc29sZS5yZWRoYXQuY29tIl0sInJlYWxtX2FjY2VzcyI6eyJyb2xlcyI6WyJhdXRoZW50aWNhdGVkIiwicG9ydGFsX21hbmFnZV9zdWJzY3JpcHRpb25zIiwiZXJyYXRhOm5vdGlmaWNhdGlvbl9zdGF0dXNfZW5hYmxlZCIsImVycmF0YTpub3RpZmljYXRpb246ZW5oYW5jZW1lbnQiLCJwb3J0YWxfbWFuYWdlX2Nhc2VzIiwiZXJyYXRhOm5vdGlmaWNhdGlvbjpzZWN1cml0eSIsImVycmF0YTpub3RpZmljYXRpb25fbGV2ZWxfc3lzdGVtLXZpc2libGUiLCJlcnJhdGE6bm90aWZpY2F0aW9uOmJ1Z2ZpeCIsIm9mZmxpbmVfYWNjZXNzIiwiYWRtaW46b3JnOmFsbCIsInVtYV9hdXRob3JpemF0aW9uIiwicG9ydGFsX3N5c3RlbV9tYW5hZ2VtZW50IiwiZXJyYXRhOm5vdGlmaWNhdGlvbl9kZWxpdmVyeV9pbnN0YW50IiwicmhkX2FjY2Vzc19taWRkbGV3YXJlIiwicG9ydGFsX2Rvd25sb2FkIl19LCJyZXNvdXJjZV9hY2Nlc3MiOnsicmhkLWRtIjp7InJvbGVzIjpbInJodXNlciJdfSwiYWNjb3VudCI6eyJyb2xlcyI6WyJtYW5hZ2UtYWNjb3VudCIsIm1hbmFnZS1hY2NvdW50LWxpbmtzIiwidmlldy1wcm9maWxlIl19fSwic2NvcGUiOiJvcGVuaWQgcHJvZmlsZSBlbWFpbCBhcGkuaWFtLnNlcnZpY2VfYWNjb3VudHMiLCJzaWQiOiI5YTkwNmI5My0zNjUwLTRkODEtYWEzZC0zZDQwOWZmMWE5ZTYiLCJhY2NvdW50X251bWJlciI6IjEyMzQ1Njc4IiwiaXNfaW50ZXJuYWwiOmZhbHNlLCJlbWFpbF92ZXJpZmllZCI6dHJ1ZSwicHJlZmVycmVkX3VzZXJuYW1lIjoiaHdvcmxkIiwibG9jYWxlIjoiZW5fVVMiLCJnaXZlbl9uYW1lIjoiSm9obiIsImlzX29yZ19hZG1pbiI6dHJ1ZSwiYWNjb3VudF9pZCI6IjEyMzQ1Njc4Iiwib3JnX2lkIjoiMTIzNDU2NzgiLCJyaC11c2VyLWlkIjoiMTIzNDU2NzgiLCJyaC1vcmctaWQiOiIxMjM0NTY3OCIsIm5hbWUiOiJIZWxsbyIsImZhbWlseV9uYW1lIjoiV29ybGQiLCJlbWFpbCI6Imh3b3JsZEByYW5kb20uY29tIn0.LSp4CuYBFe6ovHUT1H992M3oPPJb-1sjG6np-08kd8M",
+					RefreshToken: "valid",
+				},
+				connection: &kcconnection.Connection{
+					Token: &token.Token{
+						AccessToken:  "valid",
+						RefreshToken: "valid",
+					},
+				},
+			},
+			expectedUserName: "hellow",
+			wantErr:          true,
+		},
+	}
+	for _, tt := range tests {
+		tt.args.connection.Config = mockutil.NewConfigMock(tt.args.cfg)
+
+		loggerBuilder := logging.NewStdLoggerBuilder()
+		loggerBuilder = loggerBuilder.Debug(true)
+		logger, _ := loggerBuilder.Build()
+		localizer, _ := goi18n.New(nil)
+
+		t.Run(tt.name, func(t *testing.T) {
+
+			writerStream := iostreams.System()
+
+			buf := bytes.NewBufferString("")
+			writerStream.Out = buf
+
+			fact := &factory.Factory{
+				IOStreams: writerStream,
+				Config:    mockutil.NewConfigMock(tt.args.cfg),
+				Connection: func() (connection.Connection, error) {
+					return mockutil.NewConnectionMock(tt.args.connection, nil), nil
+				},
+				Localizer: localizer,
+				Logger:    logger,
+			}
```

Comment:
Is there a better way to reduce the boilerplate to run a command in each unit test?

Reply:
Sounds good 👍  yes factory will probably need to be configured independently for each test